File: algorithm/src/baekjoon/java/BaekJoon1253.java
package baekjoon.java;

import java.io.*;
import java.util.Arrays;
import java.util.StringTokenizer;

public class BaekJoon1253 {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        int n = Integer.parseInt(br.readLine());

        StringTokenizer st = new StringTokenizer(br.readLine());
        long[] a = new long[n];
        for (int i = 0; i < n; i++) {
            a[i] = Long.parseLong(st.nextToken());
        }
        Arrays.sort(a);

        int count = 0;

        for (int i = 0; i < n; i++) {
            int leftIndex = 0;
            int rightIndex = n-1;
            while (leftIndex < rightIndex) {
                if (a[leftIndex] + a[rightIndex] > a[i]) {
                    rightIndex--;
                } else if (a[leftIndex] + a[rightIndex] < a[i]) {
                    leftIndex++;
                } else {
                    if (leftIndex != i && rightIndex != i) {
                        count++;
                        break;
                    } else if (leftIndex == i) {
                        leftIndex++;
                    } else if (rightIndex == i) {
                        rightIndex--;
                    }
                }
            }
        }

        bw.write(String.valueOf(count));
        bw.flush();
    }
}

File: algorithm/src/baekjoon/kotlin/BaekJoon1253.kt
package baekjoon.kotlin

import java.util.StringTokenizer

fun main() {
    val n = readLine()!!.toInt()
    val a = IntArray(n)
    val st = StringTokenizer(readLine())

    for (i in 0 until n) {
        a[i] = st.nextToken().toInt()
    }
    a.sort()

    var result = 0
    for (i in 0 until n) {
        var k = 0
        var l = n-1
        while (k < l) {
            if (k == i) {
                k++
                continue
            } else if (l == i) {
                l--
                continue
            }
            val sum = a[k] + a[l]
            if (sum == a[i]) {
                result++
                break
            } else if (sum > a[i]) {
                l--
            } else {
                k++
            }
        }
    }
    println(result)
}